src/app/redux/history/reducer.js

```diff
@@ -0,0 +1,51 @@
+export const initialState = {
+  history: [],
+};
+
+export function historyReducer(state = initialState, action) {
+  switch (action.type) {
+    case 'GET_HISTORIES':
+      return state.history;
+    case 'SET_HISTORY':
+      return {
+        ...state,
+        history: [...state.history, action.payload],
+      };
+    case 'EDIT_EXPRESSION_HISTORY': {
+      let {newExpression, id} = action.payload;
+      let historyCopy = [];
+      historyCopy = [...state.history];
+      let historyFiltered = [];
+      historyCopy.map((expression, key) => {
+        if (key !== id) {
+          historyFiltered.push(expression);
+        } else {
+          historyFiltered.push(newExpression);
+        }
+      });
```

Comment:
A better approach would be to do a findIndex for the expression to edit, and then just do `historyCopy[index] = newExpression`.

then `history: historyCopy`

Reply:
Understand, in this case the id that comes is the position directly inside the array. So i could assign it `historyCopy[id] = newExpression`. Already made the changes.